instrumentation/manager.go

```diff
@@ -0,0 +1,348 @@
+package instrumentation
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/go-logr/logr"
+
+	"github.com/odigos-io/odigos/instrumentation/detector"
+)
+
+var (
+	errNoInstrumentationFactory = errors.New("no ebpf factory found")
+	errFailedToGetDetails       = errors.New("failed to get details for process event")
+	errFailedToGetDistribution  = errors.New("failed to get otel distribution for details")
+	errFailedToGetConfigGroup   = errors.New("failed to get config group")
+)
+
+// ConfigUpdate is used to send a configuration update request to the manager.
+// The manager will apply the configuration to all instrumentations that match the config group.
+type ConfigUpdate[configGroup ConfigGroup] map[configGroup]Config
+
+type instrumentationDetails[details Details, configGroup ConfigGroup] struct {
```

Comment:
Since we already have `Details`, and now `instrumentationDetails` with a property `details` inside it, can we have better names so it's less confusing?

This creates code like `details.details`

Reply:
Renamed `Details` to `ProcessGroup` and added comments explaining the meaning behind it